crates/valence_server/src/status_effect.rs

```diff
@@ -0,0 +1,152 @@
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use bevy_ecs::query::WorldQuery;
+use valence_entity::active_status_effects::{ActiveStatusEffect, ActiveStatusEffects};
+use valence_entity::entity::Flags;
+use valence_entity::living::{PotionSwirlsAmbient, PotionSwirlsColor};
+use valence_protocol::packets::play::{
+    entity_status_effect_s2c, EntityStatusEffectS2c, RemoveEntityStatusEffectS2c,
+};
+use valence_protocol::status_effects::StatusEffect;
+use valence_protocol::{VarInt, WritePacket};
+
+use crate::client::Client;
+use crate::EventLoopPostUpdate;
+
+pub struct StatusEffectPlugin;
+
+impl Plugin for StatusEffectPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            EventLoopPostUpdate,
+            (
+                add_status_effects,
+                update_active_status_effects,
+                add_status_effects, // only expired effects should be here
+            ),
+        );
+    }
+}
+
+fn update_active_status_effects(mut query: Query<&mut ActiveStatusEffects>) {
+    for mut active_status_effects in query.iter_mut() {
+        active_status_effects.increment_active_ticks();
+    }
+}
+
+fn create_packet(effect: &ActiveStatusEffect) -> EntityStatusEffectS2c {
+    EntityStatusEffectS2c {
+        // everywhere else in the codebase, this is the player's entity id
+        // will probably need to change this later
+        entity_id: VarInt(0),
```

Comment:
This needs to be addressed

Reply:
Out of the scope of this pr tho, kuz we'd have to change it everywhere

Reply:
I see that entity id of zero means client id as mentioned here: https://github.com/valence-rs/valence/blob/14cc9bdeb2b82e71e52882f65285c07e20e8bf6a/crates/valence_server/src/spawn.rs#L113

So maybe it's fine to leave zero here.

Reply:
Ah ok, then lets tweak the comments so that's more clear.

Reply:
Changed it :+1: